convert_to_hf.py

```diff
@@ -0,0 +1,209 @@
+    linear_weights_not_to_quantize=None,
+    current_key_name=None,
+):
+    for name, module in model.named_children():
+        if current_key_name is None:
+            current_key_name = []
+        current_key_name.append(name)
+
+        if isinstance(module, QuantizedLinear):
+            # Check if the current key is not in the `linear_weights_not_to_quantize`
+            if ".".join(current_key_name) + ".weight" not in linear_weights_not_to_quantize:
+                tensor_name = '.'.join(current_key_name)
+                quantization_config_shapes[f'{tensor_name}.dense_weights.shape'] = module.dense_weights.shape[0]
+                quantization_config_shapes[f'{tensor_name}.row_offsets.shape'] = module.row_offsets.shape[0]
+                quantization_config_shapes[f'{tensor_name}.col_vals.shape'] = module.col_vals.shape[0]
+                quantization_config_shapes[f'{tensor_name}.in_perm.shape'] = module.in_perm.shape[0]
+                # Store the module class in case we need to transpose the weight later
+                model._modules[name].source_cls = type(module)
+                # Force requires grad to False to avoid unexpected errors
+                model._modules[name].requires_grad_(False)
+        if len(list(module.children())) > 0:
+            _, has_been_replaced = replace_with_spqr_linear(
+                module,
+                quantization_config_shapes=quantization_config_shapes,
+                linear_weights_not_to_quantize=linear_weights_not_to_quantize,
+                current_key_name=current_key_name,
+            )
+        # Remove the last key for recursion
+        current_key_name.pop(-1)
+    return model, True
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser(add_help=True)
+
+    parser.add_argument(
+        "--model",
+        type=str,
+        help="Path to the model to base config on, as in AutoConfig.from_pretrained()",
+    )
+    parser.add_argument(
+        "--config_path",
+        type=str,
+        help="Path to the model to base config on, as in AutoConfig.from_pretrained()",
+    )
+    parser.add_argument(
+        "--in_path_pt",
+        type=str,
+        help="Path of the checkpoint to convert",
+    )
+    parser.add_argument(
+        "--out_path",
+        type=str,
+        help="Path to save HF compatible checkpoint to",
+    )
+    parser.add_argument(
+        "--save_safetensors",
+        action="store_true",
+        help="Whether to save in safetensors format",
+    )
+    parser.add_argument(
+        "--trust_remote_code",
+        action="store_true",
+        help="Whether to trust remote code",
+    )
+    parser.add_argument(
+        "--load_model",
+        action="store_true",
+        help="Whether to load model",
+    )
+    parser.add_argument(
+        "--save_tokenizer",
+        action="store_true",
+        help="Whether to save tokenizer",
+    )
+    args = parser.parse_args()
+
```

Comment:
Please specify in Readme what kind of model should be inputed here to be converted into HF format? Is it from legacy or your format of SPQR. Ideally some script -  example how to do it.